.github/workflows/ci.yaml

```
@@ -0,0 +1,45 @@
+name: CI
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - macos-latest
+        go:
+          - 1.16
+    runs-on: ${{ matrix.os }}
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      -
+        name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go }}
+      -
+        name: Cache Go modules
+        uses: actions/cache@v1
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      -
+        name: Tests
+        run: |
+          go mod tidy
+          diff <(gofmt -d .) <(printf "")
+          go test -v ./...
+          go test -cpu=1 -short -race ./...
+          go test -cpu=4 -short -race ./...
```

Tests if goreleaser can run:
```suggestion
          go test -cpu=4 -short -race ./...
          curl -sfL https://git.io/goreleaser | VERSION=v1.2.5 sh -s -- check
          curl -sL https://git.io/goreleaser | VERSION=v1.2.5 sh -s -- --snapshot --skip-publish --rm-dist
```

---

But this is going to fail on Windows for sure

---

@TheSAS You are right. You can do these on a single os/go version combination, for example:

https://github.com/klauspost/compress/blob/master/.github/workflows/go.yml#L42

---

go-realeaser is running later in another workflow.

---

@TheSAS You want to check if releases can be built in PRs. Otherwise you will only see the problems when you actually do a release.

---

Aha, got it, thank you for the explanation @klauspost